engine/runtime/src/test/scala/org/enso/interpreter/test/instrument/RuntimeServerTest.scala

```diff
@@ -441,6 +529,108 @@ class RuntimeServerTest
           Vector(Api.ExpressionValueUpdate(idMain, Some("Unit"), None))
         )
       ),
+      Api.Response(
+        Api.SuggestionsDatabaseReIndexNotification(
+          moduleName,
+          Seq(
+            Api.SuggestionsDatabaseUpdate.Add(
+              Suggestion.Method(
+                None,
+                moduleName,
+                "foo",
+                Seq(
+                  Suggestion.Argument("this", "Any", false, false, None),
+                  Suggestion.Argument("a", "Any", false, false, None),
+                  Suggestion.Argument("b", "Any", false, false, None)
+                ),
+                "here",
+                "Any",
+                None
+              )
+            ),
+            Api.SuggestionsDatabaseUpdate.Add(
+              Suggestion.Method(
+                Some(idMain),
+                moduleName,
+                "main",
+                List(Suggestion.Argument("this", "Any", false, false, None)),
+                "here",
+                "Any",
+                None
+              )
+            )
+          )
+        )
+      )
+    )
+    context.consumeOut shouldEqual List("3")
+  }
+
+  it should "perform operations with State" in {
+    val contextId  = UUID.randomUUID()
+    val requestId  = UUID.randomUUID()
+    val moduleName = "Test.Main"
+
+    val metadata  = new Metadata
+    val idMain    = metadata.addItem(7, 27)
+    val idMainBar = metadata.addItem(26, 8)
+
+    val code =
```

Comment:
Please extend the testing scenario, by entering `bar` and verifying that state is indeed correct in a frame detached from the original call. Moreover, pass something non-trivial (I recommend 42) to `State.run` and then start `bar` with a `State.get` and verify that it indeed returned a 42.